src/redux/books/books.js

```diff
@@ -74,4 +74,4 @@ const reducer = (state = initialState, action) => {
   }
 };
 
```

Comment:
- [ ] Kindly let's create the `addBook and removeBook` action creators and export them as mentioned in the project requirements below. with this you will no longer have to dispatch the actions `REMOVE_BOOK and ADD_BOOK` directly from the `removeBook and addBook` thunk creators. see screenshot below 👇 

![image](https://user-images.githubusercontent.com/55465133/135726051-6a2b1127-d226-42c1-8385-3ad5328a1488.png)

Reply:
Dear @barackm,

I did it as you marked, please could you see the lines 39-40 in books.js

Here is also a screenshot of my code:
![image](https://user-images.githubusercontent.com/49064106/135726514-839f54b0-9176-418b-8a61-d61159d58cc2.png)

Reply:
Hi @elmar8287,
Yeah I saw that, but as you can see, you are trying to dispatch an object that has the property `type` set to `ADD_BOOK` but this is not actually an action creator, an action creator should look like this 👇 
```
const addBook = (payload) => ({
    type: ADD_BOOK,
    payload,
});

const addBookAsync = (payload) => async (dispatch) => {
------- your code for adding a new book goes here -----
if(adding) {
    dispatch(addBook(payload))   
}
}
```

as you can see now I am dispatching an action creator instead of an action directly as what you had before 😉